src/main/java/com/foo/gosucatcher/domain/image/infrastructure/FileSystemExpertImageService.java

```diff
@@ -0,0 +1,187 @@
+			Files.copy(inputStream, root.resolve(newFilename), StandardCopyOption.REPLACE_EXISTING);
+		} catch (IOException e) {
+			throw new BusinessException(ErrorCode.INTERNAL_SERVER_ERROR);
+		}
+
+		return newFilename;
+	}
+
+	private Expert findExpert(Long id) {
+
+		return expertRepository.findById(id)
+			.orElseThrow(() -> new EntityNotFoundException(ErrorCode.NOT_FOUND_EXPERT));
+	}
+
+	private void saveExpertImage(String filename, MultipartFile file, Expert expert, Path root) {
+		ExpertImage expertImage = ExpertImage.builder()
+			.filename(filename)
+			.url(root.resolve(filename).toString())
+			.size(file.getSize())
+			.expert(expert)
+			.build();
+
+		expertImageRepository.save(expertImage);
+	}
+
+	private void throwEntityNotFoundException() {
+		throw new EntityNotFoundException(ErrorCode.NOT_FOUND_IMAGE);
+	}
+
+	@Override
+	public String store(ImageUploadRequest request) {
+		MultipartFile file = validateFile(request.file());
+
+		Path root = ensureDirectoryExists(request.id());
+
+		String newFilename = saveFile(file, root);
+
+		Expert expert = findExpert(request.id());
+
+		saveExpertImage(newFilename, file, expert, root);
+
+		return newFilename;
+	}
+
+	@Override
+	public List<ImageResponse> loadAll(Long expertId) {
+		try {
+			Path root = Paths.get(uploadPath, expertId.toString());
+			return Files.walk(root, 1)
+				.filter(path -> !path.equals(root))
+				.map(path -> {
+					String filename = path.getFileName().toString();
+					String url = MvcUriComponentsBuilder.fromMethodName(ExpertController.class,
+						"getImage", expertId, filename).build().toString();
+					Long size;
+					try {
+						size = Files.size(path);
+					} catch (IOException e) {
+						size = 0L;
+					}
+					return new ImageResponse(filename, url, size);
+				})
+				.collect(Collectors.toList());
+		} catch (IOException e) {
+			throw new ExpertImageIOException(ErrorCode.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	@Override
+	public Path load(Long expertId, String filename) {
+
+		return Paths.get(uploadPath, expertId.toString()).resolve(filename);
+	}
+
+	@Override
+	public Resource loadAsResource(Long expertId, String filename) {
+		try {
+			Path file = load(expertId, filename);
+			Resource resource = new UrlResource(file.toUri());
+			if (resource.exists() || resource.isReadable()) {
+				return resource;
+			}
+			throw new ExpertImageIOException(ErrorCode.INVALID_IMAGE);
+
+		} catch (MalformedURLException e) {
+			throw new ExpertImageIOException(ErrorCode.INVALID_IMAGE);
+		}
+	}
+
+	@Override
+	public void delete(Long expertId, String filename) {
+		try {
+			Path root = Paths.get(uploadPath, expertId.toString());
+
+			if (!Files.exists(root)) {
+				throw new EntityNotFoundException(ErrorCode.NOT_FOUND_IMAGE);
+			}
+
+			Optional<ExpertImage> expertImage = expertImageRepository.findByFilename(filename);
+
+			expertImage.ifPresentOrElse(
+				expertImageRepository::delete,
+				this::throwEntityNotFoundException
+			);
```

Comment:
이 클래스 중간 부분의 throwEntityNotFoundException 메서드는 지우셔도 될 것 같습니다! 제가 아래에 적은 코드는 깔끔한 코드는 아니지만,,,

```suggestion
			expertImage.ifPresentOrElse(
				expertImageRepository::delete,
				() -> {
				    throw new EntityNotFoundException(ErrorCode.NOT_FOUND_IMAGE);
				}
			);
```
이정도로 바꿀 수 있을 것 같네요ㅎㅎ